### ISSUE TITLE
Add feature to check if a mobs hitbox can fit at a given position

### ISSUE THREAD

**Issue by ElCeejo** [minetest/minetest#12590]
Low priority but would be incredibly helpful for pathfinding and other mob traversal methods. Currently the best options are to, take 2 positions representing far corners of a hitbox and use a raycast at all positions in the area, or checking if the node at the position is solid (walkable == true). These aren't optimal, as doing it accurately with raycasts is costly and checking if the node is walkable doesn't work well for slabs or doors.

The best solution I can think of is a function that takes 2 positions and checks if the area is solid or not.

**Comment by appgurueu** [comment]
> Currently the best options are to, take 2 positions representing far corners of a hitbox and use a raycast at all positions in the area, or checking if the node at the position is solid (walkable == true).

Wrong. The best option currently is to use `modlib.minetest.get_node_collisionboxes` for a couple nearby nodes, then check whether they intersect using `modlib.vector.box_box_intersection`. You can find some similar code [here](https://github.com/appgurueu/disable_build_where_they_stand/blob/20443fb02f3b54b4830c953bcc4ffcdbdd70691d/init.lua#L26-L37).
